.github/workflows/ci.yml

```diff
@@ -1,34 +1,39 @@
 name: CI
 
-on: [push, pull_request]
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+  push:
+    branches: [master]
 
 env:
   CARGO_TERM_COLOR: always
-  
+
 jobs:
   rustfmt:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - uses: actions-rs/toolchain@v1
+      - name: install Rust
+        uses: actions-rust-lang/setup-rust-toolchain@v1.4.3
```

Comment:
Are there any benefits if we turn to use `actions-rust-lang`?

Reply:
The concensus is that actions-rs is unmaintained and therefore soft-deprecated. See https://github.com/actions-rs/toolchain/issues/216.

The upside is that this action set up includes automatic target caching and sensible defaults like disabling incremental compilation and using sparse registries, improving the feedback time of CI runs significantly.

Reply:
There isn't yet an alternative for `actions-rs/clippy-check`, hence why it remains.

Reply:
> There isn't yet an alternative for `actions-rs/clippy-check`, hence why it remains.

Why don't we just run `cargo clippy --all-targets --all-features -- -D warnings`?

Reply:
This check annotates PRs with the problems it finds.

Reply:
I checked [the log](https://github.com/magiclen/cidr-utils/actions/runs/4527727498/jobs/7976555360) and found `actions-rs/clippy-check` seems not to work properly.

Reply:
It doesn't work from forks. It's a reasonable security measure.

Reply:
When does it work? Any example?

Reply:
It works on source branches in this repo. [The readme](https://github.com/actions-rs/clippy-check#rust-clippy-check-action) has a screenshot of what it would look like in PR diffs.

Reply:
I have seen that before. But I have no idea how to reproduce it.

Reply:
Since the value is more limited here, I've changed it to the command you recommended.